core/src/test/java/fr/sncf/osrd/envelope/EnvelopeOverlayThresholdTest.java

```diff
@@ -34,6 +34,26 @@ public void testSingleStepCrossLowerThreshold() {
         EnvelopeTransitions.checkContinuity(envelope, true, false);
     }
 
+    @Test
+    public void testThresholdIntersectionInPartTransition() {
+        var constSpeedEnvelopes = Envelope.make(
+                EnvelopePart.generateTimes(
+                        null,
+                        new double[]{0, 10},
+                        new double[]{100, 100}
+                ),
+                EnvelopePart.generateTimes(
+                        null,
+                        new double[]{10, 20},
+                        new double[]{100, 100}
+                )
+        );
+        var cursor = EnvelopeCursor.forward(constSpeedEnvelopes);
+        var builder = new OverlayEnvelopeBuilder(cursor);
+        var partBuilder = builder.startContinuousOverlay(null, 95, CmpOperator.LOWER);
+        partBuilder.addStep(15, 90);
```

Comment:
Why is this test not asserting anything ?

Reply:
It only tests that no internal assert is triggered inside of `addStep`. It failed when I wrote this test.